### PR TITLE
rainerscript: add split() function

### PR DESCRIPTION
**Pull request by 20syldev**
### Summary
Users need to parse delimited strings (CSV, tags, paths) into arrays for iteration or JSON output without external processing.

### Notes
- Tests included: `rscript_split.sh` and `rscript_split-vg.sh`
- Documentation added: `doc/source/rainerscript/functions/rs-split.rst`

Impact: New RainerScript function available to all users.
- **Before:** No native way to split strings into arrays in RainerScript. 
- **After:** `split(string, separator)` returns a JSON array of substrings.

### CODE REVIEW

**Comment by 20syldev** [comment]
> ## Code Review
> This pull request introduces a new `split()` function to RainerScript, which is a great addition for parsing delimited strings. The implementation is mostly correct, but I've found a critical issue regarding inconsistent behavior between single-character and multi-character separators, especially with edge cases like leading, trailing, or multiple consecutive delimiters. My review includes a suggestion to unify the logic for a more predictable and robust `split()` function. The documentation and tests are a good start for this new feature.

Gemini's fix might look like this:

```c
static void ATTR_NONNULL() doFunct_split(struct cnffunc *__restrict__ const func,
                                        struct svar *__restrict__ const ret,
                                        void *const usrptr,
                                        wti_t *const pWti) {
    struct svar srcVal[2];
    int bMustFree = 0;
    int bMustFree2 = 0;

    cnfexprEval(func->expr[0], &srcVal[0], usrptr, pWti);
    cnfexprEval(func->expr[1], &srcVal[1], usrptr, pWti);

    char *inputStr = (char *)var2CString(&srcVal[0], &bMustFree);
    char *separator = (char *)var2CString(&srcVal[1], &bMustFree2);

    struct json_object *jsonArray = json_object_new_array();

    if (jsonArray == NULL || inputStr == NULL || separator == NULL || strlen(separator) == 0) {
        goto done;
    }

    if (*inputStr == '\0') {
        goto done;
    }

    char *workStr = strdup(inputStr);
    if (workStr == NULL) {
        goto done;
    }

    const size_t sepLen = strlen(separator);
    char *p = workStr;

    while (1) {
        char *next_sep = strstr(p, separator);
        if (next_sep != NULL) {
            *next_sep = '\0';
            json_object_array_add(jsonArray, json_object_new_string(p));
            p = next_sep + sepLen;
        } else {
            json_object_array_add(jsonArray, json_object_new_string(p));
            break;
        }
    }

    free(workStr);

done:
    ret->datatype = 'J';
    ret->d.json = jsonArray;

    if (bMustFree) free(inputStr);
    if (bMustFree2) free(separator);
    varFreeMembers(&srcVal[0]);
    varFreeMembers(&srcVal[1]);
}
```

File: tests/Makefile.am [hunk 1]
```diff
@@ -623,6 +624,7 @@ TESTS +=  \
 	threadingmqaq.sh \
 	func-substring-invld-startpos-vg.sh \
 	rscript_trim-vg.sh
+	rscript_split-vg.sh \
```
Comment (rgerhards) [inline tests/Makefile.am]: The backslash is in the wrong line, it needs to go into the one above. As it, the line is continued into the next if. This breaks the build as can be seen in CI.

Comment (rgerhards) [reply tests/Makefile.am]: note that i did not check for all occurrences, if there are multiple.

**Comment by rgerhards** [comment]
Thx for the PR, it looks useful once the issues are solved. If you need help, pls let me know.

File: tests/Makefile.am (hunk 1 above)
Comment (20syldev) [reply tests/Makefile.am]: I'll fix that
